### ISSUE TITLE
Different lists of examples when shuffle == False

### ISSUE THREAD

**Issue by experiencor** [mosaicml/streaming#749]
OS: [macOS]
mosaicml-streaming==0.7.6

## To reproduce

from streaming import StreamingDataset, Stream

```
streams = [
    Stream(
        local="/tmp/dataset",
        remote='s3://path/to/dataset',
        choose=10,
    ),
]

dataset = StreamingDataset(
    streams=streams,
    shuffle=False,
    batch_size=1,
    predownload=8
)

for example in dataset:
    print(example["id"])

print("=" * 10)

for example in dataset:
    print(example["id"])
```

## Expected behavior

The same list of examples for 2 iterations of dataset when shuffle = False.

## Actual behavior
Different list of examples for 2 interations of dataset.

![Screenshot 2024-08-13 at 6 23 11 PM](https://github.com/user-attachments/assets/dabfaefe-1ca9-4f0d-b3b6-5eae6f130e30)

**Comment by XiaohanZhangCMU** [comment]
@experiencor with "choose' in the stream, StreamingDataset will do upsample/downsample per stream. The random seed depends on epochs. So when you compare 1st epoch with 2nd epoch, they will be different. 
The randomness comes from [here](https://github.com/mosaicml/streaming/blob/260233048936c744691dea362f115aae0055b26e/streaming/base/sampling.py#L64)

**Comment by snarayan21** [comment]
@experiencor Thanks for filing this issue! Fixed in #750.
